sno/working_copy/sqlserver_adapter.py

```diff
@@ -79,7 +108,51 @@ def v2_schema_to_sqlserver_spec(schema, v2_obj):
     return ", ".join(result)
 
 
-def v2_type_to_ms_type(column_schema, v2_obj):
+def v2_column_schema_to_sqlserver_spec(column_schema, v2_obj):
+    name = column_schema.name
+    ms_type = v2_type_to_ms_type(column_schema)
+    constraints = []
+
+    if ms_type == "geometry":
+        extra_type_info = column_schema.extra_type_info
+        geometry_type = extra_type_info.get("geometryType")
+        if geometry_type is not None and geometry_type != "GEOMETRY":
+            constraints.append(_geometry_type_constraint(name, geometry_type))
+
+        crs_name = extra_type_info.get("geometryCRS")
+        crs_id = None
+        if crs_name is not None:
+            crs_id = crs_util.get_identifier_int_from_dataset(v2_obj, crs_name)
+        if crs_id is not None:
+            constraints.append(_geometry_crs_constraint(name, crs_id))
+
+    if constraints:
+        constraint = f"CHECK({' AND '.join(constraints)})"
+        return " ".join([quote(column_schema.name), ms_type, constraint])
+
+    return " ".join([quote(column_schema.name), ms_type])
+
+
+def _geometry_type_constraint(col_name, geometry_type):
+    result = f"({quote(col_name)}).STGeometryType()"
+
+    geometry_type = geometry_type.upper()
+    geometry_types = _GEOMETRY_SUBTYPES.get(geometry_type)
+
+    if geometry_types:
+        geometry_types = ",".join(f"'{g}'" for g in geometry_types)
+        result += f" IN {geometry_types}"
```

Comment:
[The docs](https://docs.microsoft.com/en-us/sql/t-sql/spatial-geometry/stgeometrytype-geometry-data-type?view=sql-server-ver15) and some examples on the web seem to have `StGeometryType()` returning `'LineString'` instead of `'LINESTRING'`. 

I guess if the tests pass then the internal comparison is case-insensitive? 🤷

Reply:
Ugh. I did wonder about this... tests pass... but it might depend on the collation settings of my SQL Server instance, which of course, shouldn't be relied upon.
Fixing this now.